pyscriptjs/tests/integration/test_plugins.py

```diff
@@ -173,6 +209,28 @@ def test_execution_hooks(self):
 
         # TODO: It'd be actually better to check that the events get called in order
```

Comment:
Ah, so that's actually a note on `test_execution_hooks()` that predates this PR - I think it's from when Antonio originally wrote the test? Basically, that test checks that each of the plugin methods were called, but doesn't actually verify that they're in called in the appropriate order. I didn't make that any better, just added more methods to be checked for.